pilot/pkg/serviceregistry/kube/controller/endpointslice.go

```diff
@@ -182,24 +182,21 @@ func (esc *endpointSliceController) serviceTargets(ep *v1.EndpointSlice, proxy *
 				log.Warnf("unexpected state, svc %v missing port %v", svc.Hostname, instance.ServicePortName)
 				continue
 			}
-			// consider multiple IP scenarios
-			for _, ip := range proxy.IPAddresses {
-				if ip != instance.Address {
-					continue
-				}
-				// If the endpoint isn't ready, report this
-				if instance.HealthStatus == model.UnHealthy && esc.c.opts.Metrics != nil {
-					esc.c.opts.Metrics.AddMetric(model.ProxyStatusEndpointNotReady, proxy.ID, proxy.ID, "")
-				}
-				si := model.ServiceTarget{
-					Service: svc,
-					Port: model.ServiceInstancePort{
-						ServicePort: port,
-						TargetPort:  instance.EndpointPort,
-					},
-				}
-				out = append(out, si)
+			if proxy.IPAddresses[0] != instance.FirstAddressOrNil() {
```

Comment:
same comment as the listener code - how do we know these will be ordered the same?

Reply:
See my comment https://github.com/istio/istio/pull/51279#discussion_r1677684709